### ISSUE TITLE
News updates are delayed sometimes

### ISSUE THREAD

**Issue by ShwangShwing** [ShwangShwing/sofia-swing-dance-festival-android#24]
Sometimes a user will receive push notification for news, but he won't see the news for sometime.
Possible reason is that the device goes to sleep and the connection to the firebase realtime database becomes bugged. Research a way to precache the new news article when the push notification is received.

**Comment by ShwangShwing** [comment]
This was most probably caused by a bug in the firebase library. I found a way to reproduce the issue on virtual device:

Name: Pixel_API_25 CPU/ABI: Google APIs Intel Atom (x86) Path: /home/shwangshwing/.android/avd/Pixel_API_25.avd Target: google_apis [Google APIs] (API level 25) Skin: 1080x1920 SD Card: 100 MB hw.dPad: no

To reproduce it I did the following. 
1. Connected the device to the mobile network only (no Wi-Fi)
2. Opened the my events page with event inside that interests me.
3. Changed the event in the db and observed the event change in the app.
4. Stopped the mobile network connection (the device had no access then)
5. Changed the event in the database again
6. Connect the device using the wi-fi
7. Observe how the event DOESN'T change
8. Send the app to background and to foreground. Observe how the changes in the database are not reflected in the app, suggesting that the firebase connection was broken.
